sharding-jdbc/sharding-jdbc-core/src/test/java/org/apache/shardingsphere/shardingjdbc/common/base/AbstractEncryptJDBCDatabaseAndTableTest.java

```diff
@@ -17,87 +17,90 @@
+        encryptRuleConfig = getEncryptRuleConfig(getFile("encrypt_config.yaml"));
+        System.out.println(encryptRuleConfig);
         if (null != encryptDataSource && null != encryptDataSourceWithProps) {
             return;
         }
         Map<String, DataSource> dataSources = getDataSources();
         encryptDataSource = new EncryptDataSource(dataSources.values().iterator().next(), new EncryptRule(createEncryptRuleConfiguration()), new Properties());
         encryptDataSourceWithProps = new EncryptDataSource(dataSources.values().iterator().next(), new EncryptRule(createEncryptRuleConfiguration()), createProperties());
     }
-    
+
     private static Properties createProperties() {
         Properties result = new Properties();
         result.put(ConfigurationPropertyKey.SQL_SHOW.getKey(), true);
         result.put(ConfigurationPropertyKey.QUERY_WITH_CIPHER_COLUMN.getKey(), false);
         return result;
     }
-    
+
     private static Map<String, DataSource> getDataSources() {
         return Maps.filterKeys(getDatabaseTypeMap().values().iterator().next(), ENCRYPT_DB_NAMES::contains);
     }
-    
+
     private static EncryptRuleConfiguration createEncryptRuleConfiguration() {
         EncryptRuleConfiguration result = new EncryptRuleConfiguration();
-        result.getEncryptors().put("test", new EncryptorRuleConfiguration("test", new Properties()));
-        result.getEncryptors().put("assistedTest", new EncryptorRuleConfiguration("assistedTest", new Properties()));
-        result.getTables().put("t_encrypt", createEncryptTableRule());
-        result.getTables().put("t_query_encrypt", createQueryEncryptTableRule());
-        result.getTables().put("t_encrypt_contains_column", createEncryptContainsColumnTableRule());
+        for (Map.Entry<String, YamlEncryptorRuleConfiguration> entry : encryptRuleConfig.getEncryptors().entrySet()) {
+            result.getEncryptors().put(entry.getKey(), new EncryptorRuleConfiguration(entry.getValue().getType(), entry.getValue().getProps()));
+        }
+
+        for (Map.Entry<String, YamlEncryptTableRuleConfiguration> tableMap : encryptRuleConfig.getTables().entrySet()) {
+            Map<String, EncryptColumnRuleConfiguration> columns = new LinkedHashMap<>(2, 1);
+            for (Map.Entry<String, YamlEncryptColumnRuleConfiguration> columnMap : tableMap.getValue().getColumns().entrySet()) {
+                YamlEncryptColumnRuleConfiguration yamlConfig = columnMap.getValue();
+                EncryptColumnRuleConfiguration columnRuleConfiguration = new EncryptColumnRuleConfiguration(yamlConfig.getPlainColumn(), yamlConfig.getCipherColumn(),
+                    yamlConfig.getAssistedQueryColumn(),
+                    yamlConfig.getEncryptor());
+                columns.put(columnMap.getKey(), columnRuleConfiguration);
+            }
+            result.getTables().put(tableMap.getKey(), new EncryptTableRuleConfiguration(columns));
+        }
+
         return result;
     }
```

Comment:
You may use `YamlEncryptDataSourceFactory.createDataSource()` to load YAML directly.
We can just make sure the format of YAML file are correct and then use ShardingSphere's public API to load yaml and create EncryptDataSource.

Reply:
@terrymanu  Modified as per comment. Pls check.